src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -886,7 +880,7 @@ async Task<IEnumerable<IAccount>> ITokenCacheInternal.GetAccountsAsync(string au
                 OnBeforeAccess(args);
                 try
                 {
-                    tokenCacheItems = ((ITokenCacheInternal)this).GetAllRefreshTokens(false);
+                    tokenCacheItems = ((ITokenCacheInternal)this).GetAllRefreshTokens(filterByClientId);
```

Comment:
When this flag is true, RemoveAccount only looks at credentials created by that client ID. When FOCI is enabled, it no longer makes sense to scope by clientID, e.g. the RT can belong to several clients. 

We could strengthen this condition to say: scope by clientID if FOCI is disabled or if an FRT exists (semantically this makes sense to me - if FOCI is supported AND if you actually have an FRT, than your GetAccounts / RemoveAccount
cannot be scoped by clientID).

Reply:
I think this logic creates inconsistency between MSAL behaviors. We should have consistent API surface across all MSALs with consistent functionality. If there's a need for global signout for .NET, we should either:
1. Introduce a new API that explicitly removes account for all apps on the device
2. Update all MSALs to use this logic across platforms

Reply:
@oldalton @danieldobalian @brandwe : do we have any plans for a "sign-out for all apps on the device" API already ?

Reply:
I don't think we need a new API.

I will modify this behaviour to apply only if an FRT is in the cache. If an FRT is in the cache, then sign-out has to be global. No other MSALs support FOCI, so there is no inconsistency to talk about.

We are not trying to solve the concept of "global sing-out" here.